Composer/packages/client/src/router.tsx

```diff
@@ -138,6 +140,14 @@ const ProjectRouter: React.FC<RouteComponentProps<{ projectId: string; skillId:
     }
   }, [botProjectSpaceLoaded, rootBotProjectId]);
 
+  useEffect(() => {
+    if (botProjectSpaceLoadedState && rootBotProjectId && botName) {
+      const storedLocale = languageStorage.get(botName)?.locale;
+      setLocale(storedLocale, rootBotProjectId);
+    }
+  }, [botProjectSpaceLoaded, rootBotProjectId, botName]);
```

Comment:
The bot display name would be set when botProjectSpaceLoaded is set to true. Do we need another useEffect just for the additional botName. Feels like it could be combined with the above useEffect